src/chart/generateCategoricalChart.tsx

```diff
@@ -874,6 +874,7 @@ export const generateCategoricalChart = ({
           (axisMap && axisMap[id]) || entry.axisType === 'zAxis',
           `Specifying a(n) ${entry.axisType}Id requires a corresponding ${
             entry.axisType
+            // @ts-expect-error we should stop reading data from ReactElements
```

Comment:
💯% agree, the data has to come from somewhere though. Eventually, we refactor this to get the data from... well data instead of elements 😅. In the meantime the invariant is still useful with the expect error